scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorThread.java

```diff
@@ -97,21 +97,23 @@ private void doWork() {
 
   private void executeCurrentTask() {
     final var properties = currentTask.getActor().getContext();
-    MDC.setContextMap(properties);
+    boolean resubmit = false;
+
     MDC.put("actor-scheduler", actorThreadGroup.getSchedulerName());
```

Comment:
:question: Don't we need to remove this property when we are done with the current task?  Or is it okay because MDC is thread local and an actor thread always has the same actor scheduler?

Reply:
Actually, nice that you raised it - we can now set this only once on the thread, since it never changes.

So before, calling `setContextMap` would clear the map and replace it. This is why we had to set it all the time. We can now set it once and omit it forever. I'll do that.

Then we just put and remove whatever the actor's context map is. We do run the risk of an actor overwriting `actor-scheduler`, but I think that's minimal :sweat_smile:

Reply:
Oh yeah, that's nice :+1: